Fix processQueue finishing to early

llndqvst opened the pull request:
Fixes a bug where DownloadQueue.processQueue tries to launch the program
before the bootstrap has been completed.

~~Needed for https://github.com/MCUpdater/MCU-DownloadLib/pull/4~~

smbarbour commented:
This change would break pack installation as it removes the TaskableExecutor from being passed in.

llndqvst commented:
I'm not sure that's the case. That was the exact bug I was trying to fix, as you can see there wasn't even a TaskableExecutor passed in before, but a ThreadPoolExecutor which I believe is the root cause for the bug.

See https://github.com/MCUpdater/MCU-DownloadLib/pull/4/files#r570441814 where I added code to create the TaskableExecutor directly in DownloadQueue, since I didn't see any good reason to keep the creation of the executor in BootstrapForm.

If you prefer it that way I can update the pull request to reflect that, but please review the two changes together. I have tested both my own pack and http://files.mcupdater.com/example/SamplePack.xml and everything seems to work.

llndqvst committed:
Fix processQueue finishing to early.

Previously BootstrapForm did not provide a TaskableExecutor, but a
ThreadPoolExecutor causing DownloadQueue to crash when the bootstrap
process had not been finished.

llndqvst commented:
I’ve made some changes, and the launcher should now work properly. Please review

smbarbour reviewed (approved):
This is pretty much exactly what I'd have done to fix the issue.

Approved
